### PR TITLE
KEYCLOAK-5026 Store credentials

### PR DESCRIPTION
**Pull request by machielg**
Credentials are stored with user creation if they are present in the UserRepresentation. This is needed functionality for migration otherwise there is no way to store existing credentials via the REST API. I'm happy to receive feedback on the ticket and the change.

### CODE REVIEW

File: core/src/main/java/org/keycloak/representations/idm/CredentialRepresentation.java [hunk 1]
```diff
@@ -155,4 +155,101 @@ public void setCreatedDate(Long createdDate) {
     public void setConfig(MultivaluedHashMap<String, String> config) {
         this.config = config;
     }
+
```
Comment (stianst) [inline core/src/main/java/org/keycloak/representations/idm/CredentialRepresentation.java]: Is the hashCode/equals methods needed?

Comment (machielg) [inline core/src/main/java/org/keycloak/representations/idm/CredentialRepresentation.java]: The UserTest doesn't pass without it. The assertAdminEvents.assertEvent() in UserTest.createUser(UserRepresentation) checks the stored admin event, which can include the credentialrepresentation. It failed without the implemented equals method.

Comment (machielg) [inline core/src/main/java/org/keycloak/representations/idm/CredentialRepresentation.java]: I can try to remove it by relaxing the admin event verification. Let me know what you think. Thanks

**Comment by stianst** [comment]
Can you rebase please?

**Pull request merged**